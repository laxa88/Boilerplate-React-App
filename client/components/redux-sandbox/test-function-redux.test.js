import React from "react";
import { Provider } from "react-redux";
import * as enzyme from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import TestFunctionRedux from "./test-function-redux";

// Mock fetch() used by TestFunctionRedux > dispatch > loadSomeData
global.fetch = require("jest-fetch-mock");

// Mock redux store with thunk (adds asynchronous dispatching)
const mockStore = configureStore([thunk]);

describe("test-function-redux.js", () => {
  it("Should shallow render with mocked redux store", () => {
    const store = mockStore();

    const wrapper = enzyme.shallow(
      <Provider store={store}>
        <TestFunctionRedux customHandler={() => { console.log("WOAH"); }} />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("Should mount with mocked redux store and thunk", () => {
    const store = mockStore({});

    const wrapper = enzyme.mount(
      <Provider store={store}>
        <TestFunctionRedux customHandler={() => { console.log("WOAH"); }} />
      </Provider>,
    );

    expect(wrapper).toBeDefined();
  });
});
