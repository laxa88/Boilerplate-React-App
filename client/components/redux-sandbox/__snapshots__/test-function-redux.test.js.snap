// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-function-redux.js Should shallow render with mocked redux store 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(TestFunctionRedux)
    customHandler={[Function]}
  />,
  "nodes": Array [
    <Connect(TestFunctionRedux)
      customHandler={[Function]}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      >
        <Connect(TestFunctionRedux)
          customHandler={[Function]}
        />
      </Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(TestFunctionRedux)
            customHandler={[Function]}
          />,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(TestFunctionRedux)
          customHandler={[Function]}
        />,
        "_debugID": 2,
        "_renderedOutput": <Connect(TestFunctionRedux)
          customHandler={[Function]}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(TestFunctionRedux)
      customHandler={[Function]}
    />
  </Provider>,
}
`;
